core/vm/interpreter.go

```diff
@@ -855,6 +855,17 @@ func NewGethState(contract *Contract, memory *Memory, stack *Stack, Pc uint64) *
 	}
 }
 
+func (in *GethEVMInterpreter) StepN(state *GethState, numSteps int) bool {
+	in.returnData = state.Result
+
+	for i := 0; i < numSteps && !state.Halted; i++ {
+		if !in.Step(state) {
+			return false
+		}
+	}
+	return true
+}
+
 func (in *GethEVMInterpreter) Step(state *GethState) bool {
```

Comment:
The handling of the `ReadOnly` flag in lines 875-879 should also be moved into the new StepN function. Maybe even other CT specific things as well.